src/compiler.mjs

```diff
@@ -88,18 +62,18 @@ if (symbolsOnly) {
 
 // Side modules are pure wasm and have no JS
 assert(
-  !SIDE_MODULE || (ASYNCIFY && globalThis.symbolsOnly),
+  !SIDE_MODULE || (ASYNCIFY && symbolsOnly),
   'JS compiler should only run on side modules if asyncify is used.',
 );
 
 // Load compiler code
 
-load('modules.js');
-load('parseTools.js');
-load('jsifier.js');
+await import('./modules.mjs');
```

Comment:
Could be a follow, but could these `import()`s be a static import at the top?

Reply:
Yes, I should have mentioned that.

This doesn't work right now because some of these function depend on the settings from `settings.js` and we import those into `globalThis` manually still.   That means we cannot import these until we are done setting up the settings.

I'm not sure if static imports all happen?  But I think we can improve this one or another as a followup.